design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,260 @@
+# Kubevirt Feature Lifecycle
+
+## Summary
+
+Kubevirt requires a process with a clear policy on how features
+are introduced, evaluated and finally graduated or dropped.
+
+This proposal defines the steps and policies to follow in order
+to manage a feature (lifecycle) in Kubevirt.
+
+## Overview
+
+### Motivation
+Kubevirt has grown into a matured virtualization management solution
+with a large set of features.
+
+New features are being proposed and added regularly to its portfolio.
+
+With time, the challenge of supporting and maintaining such a large
+set of features raised the need to re-examine their relevance.
+It also raised the need to examine with more care features graduation.
+
+The Kubevirt community has tried to control the flow of features
+informally through feature-gates, similar to Kubernetes.
+However, as time passed, this has been found to be insufficient due
+to the lack of a well-defined processes and policy on how a feature
+should progress in its lifecycle.
+
+Eventually we would like to see features being evaluated carefully
+before they are introduced, while they are experimented with and
+proven to be actually in use (and useful) before graduating.
+
+### Goals
+- Define the process a feature needs to pass for graduation.
+- Define the process a feature needs to pass in order to be dropped.
+- Provide policies and rules on how to manage a feature during its
+  lifetime.
+
+### Non Goals
+- Implement enforcement tooling to keep features in sync with
+  the lifecycle rules.
+
+### Definition Of Users
+- Development contributors.
+- Cluster operators.
+
+### User Stories
+- As a Kubevirt contributor, I would like to introduce a new useful
+  feature and follow it to graduation (GA).
+- As a Kubevirt contributor, I would like to discontinue a feature
+  that has not yet reached (formal) graduation.
+- As a Kubevirt contributor, I would like to discontinue a feature
```

Comment:
This should only be the exception. In general we should commit to what we graduate.

Can we add a remark about this?